src/decrypt/qmc.ts

```diff
@@ -118,9 +132,20 @@ export class QmcDecoder {
       const sizeView = new DataView(sizeBuf.buffer, sizeBuf.byteOffset);
       const keySize = sizeView.getUint32(0, false);
       this.audioSize = this.size - keySize - 8;
+
       const rawKey = this.file.subarray(this.audioSize, this.size - 8);
       const keyEnd = rawKey.findIndex((v) => v == QmcDecoder.BYTE_COMMA);
+      if (keyEnd < 0) {
+        throw new Error('invalid key: search raw key failed');
+      }
       this.setCipher(rawKey.subarray(0, keyEnd));
+
+      const idBuf = rawKey.subarray(keyEnd + 1);
+      const idEnd = idBuf.findIndex((v) => v == QmcDecoder.BYTE_COMMA);
+      if (keyEnd < 0) {
+        throw new Error('invalid key: search song id failed');
+      }
+      this._songID = parseInt(textEnc.decode(idBuf.subarray(0, idEnd)));
```

Comment:
感觉可以做 `string` 类型（跳过解析到 `number` 的步骤）？

如果保留这个，建议加上第二个参数指定 10 进制。

```suggestion
      this._songID = parseInt(textEnc.decode(idBuf.subarray(0, idEnd)), 10);
```

Reply:
主要是观察了QQ音乐 SongID (`track_info.id`) 这个字段，使用的就是number类型，有此推断